pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -1612,6 +1627,24 @@ func TestDeploymentConfigChanged(t *testing.T) {
 									},
 								},
 							},
+							NodeAffinity: &corev1.NodeAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+									NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      controller.RemoteWorkerLabel,
+												Operator: corev1.NodeSelectorOpNotIn,
+												Values:   []string{""},
+											},
+											{
+												Key:      controller.ControllerDeploymentHashLabel,
+												Operator: corev1.NodeSelectorOpIn,
+												Values:   []string{"1"},
+											},
```

Comment:
This looks weird, but I guess you added it so that the "if the deployment template -affinity node selector expressions change ordering" test case would have at least two match expressions to work with, right?  Could you add a comment explaining that this second match expression is only here to satisfy the test case?